### DISCUSSION TITLE
[Question]: Custom AI Endpoints Ollama API key: Required but ignored  Errors when running the Ollama chat models

### DISCUSSION THREAD

**Discussion by getpoweredbyai** [danny-avila/LibreChat#3888] [Q&A]
### What is your question?

LibreChat         | 2024-09-01 21:43:28 error: Failed to fetch models from Ollama API. If you are not using Ollama directly, and instead, through some aggregator or reverse proxy that handles fetching via OpenAI spec, ensure the name of the endpoint doesn't start with `ollama` (case-insensitive). Cannot read properties of undefined (reading 'status')
LibreChat         | 2024-09-01 21:43:29 error: Failed to fetch models from Mistral API
LibreChat         | The request either timed out or was unsuccessful. Error message:
LibreChat         |  Cannot read properties of undefined (reading 'status')

I am trying to use Ollama Local, but I would also like to run in produciton also with Ollama, any assistance will be greatly appreciated. 

Trying to get your new Artifacts feature running with local models. 

### More Details

LibreChat         | 2024-09-01 21:43:28 error: Failed to fetch models from Ollama API. If you are not using Ollama directly, and instead, through some aggregator or reverse proxy that handles fetching via OpenAI spec, ensure the name of the endpoint doesn't start with `ollama` (case-insensitive). Cannot read properties of undefined (reading 'status')
LibreChat         | 2024-09-01 21:43:29 error: Failed to fetch models from Mistral API
LibreChat         | The request either timed out or was unsuccessful. Error message:
LibreChat         |  Cannot read properties of undefined (reading 'status')

### What is the main subject of your question?

Endpoints

### Screenshots

<img width="1362" alt="Screenshot 2024-09-01 at 17 50 58" src="https://github.com/user-attachments/assets/24a764d3-d323-4f51-b8a4-714b6c5a909a">

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
What's your config look like? did you see this? 

https://www.librechat.ai/docs/configuration/librechat_yaml/ai_endpoints/ollama

**Reply by getpoweredbyai** [reply]
<img width="1128" alt="Screenshot 2024-09-01 at 20 40 49" src="https://github.com/user-attachments/assets/ec3f0a4d-5740-442a-8b62-e3801cb4ad88">

I tried a few variations of what is on that link, each time I got the same errors. Here is a screenshot, maybe I am placing it in the wrong place? I am so new to coding.

**Reply by getpoweredbyai** [reply]
<img width="1728" alt="Screenshot 2024-09-01 at 20 48 26" src="https://github.com/user-attachments/assets/56460fad-7eb4-4731-aeb5-b9599c1b1892">

**Comment by getpoweredbyai** [comment]
My next plan was to run this on a server. will i have the same issues? can ollama models be added another way or already included like the other models?

**Reply by getpoweredbyai** [reply]
Figured it out, thanks!

**Reply by aawaheed** [reply]
May I know how you fixed it?
I am facing the same issue. Trying with Ollama locally and have set it up using librechat.yml however the response is empty.

**Reply by miche1an9e10** [reply]
How did you fix this? I am also facing the same issue. Having all set up according to the Librechat setup docs.

error: Failed to fetch models from Ollama API. If you are not using Ollama directly, and instead, through some aggregator or reverse proxy that handles fetching via OpenAI spec, ensure the name of the endpoint doesn't start with `ollama` (case-insensitive). Cannot read properties of undefined (reading 'status')

**Reply by boydthomson** [reply]
Yup, same problem.  Feels like I've tried everything... but obviously have not.

**Reply by aawaheed** [reply]
my librechat.yaml looks like this for ollama
 ```
 custom:
    # ollama
    - name: "Ollama"
      apiKey: "ollama"
      baseURL: "http://host.docker.internal:11434/v1/" 
      models:
        default: [
          "llama3.1",
          "HR"
        ]
        fetch: false # fetching list of models is not supported
      titleConvo: true
      titleModel: "current_model"
      summarize: false
      summaryModel: "current_model"
      forcePrompt: false
      modelDisplayLabel: "Ollama"
      addParams:
        "stop": [
            "<|start_header_id|>",
            "<|end_header_id|>",
            "<|eot_id|>",
            "<|reserved_special_token"
        ]
```

**Comment by MalchuL** [comment]
I found solution for myself. In case if you use LibreChat in docker, but install Ollama [manually](https://github.com/ollama/ollama/blob/main/docs/linux.md#manual-install).  
TLDR;
```
# Stop Ollama service
export OLLAMA_HOST=0.0.0.0:11434
ollama serve
```
in librechat.yaml:
```yaml
      baseURL: "http://host.docker.internal:11434/v1/chat/completions"  #To make container access to your host
````

By default Ollama uses default allowed hosts: 127.0.0.1 or localhost. But from docker host became other 172.21.0.1 (in my case for example) and Ollama not allow this. Try to change Ollama host to 0.0.0.0, that means allow all connections.

**Reply by abdullahalamodi** [reply]
> I found solution for myself. In case if you use LibreChat in docker, but install Ollama [manually](https://github.com/ollama/ollama/blob/main/docs/linux.md#manual-install). TLDR;
> 
> ```
> # Stop Ollama service
> export OLLAMA_HOST=0.0.0.0:11434
> ollama serve
> ```
> 
> in librechat.yaml:
> 
> ```yaml
>       baseURL: "http://host.docker.internal:11434/v1/chat/completions"  #To make container access to your host
> ```
> 
> By default Ollama uses default allowed hosts: 127.0.0.1 or localhost. But from docker host became other 172.21.0.1 (in my case for example) and Ollama not allow this. Try to change Ollama host to 0.0.0.0, that means allow all connections.

it's work, thank you
